src/test/java/dog/pawbook/logic/parser/AddProgramCommandParserTest.java

```diff
@@ -0,0 +1,105 @@
+package dog.pawbook.logic.parser;
+
+import static dog.pawbook.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static dog.pawbook.logic.commands.CommandTestUtil.INVALID_NAME_DESC;
+import static dog.pawbook.logic.commands.CommandTestUtil.INVALID_SESSION_DESC;
+import static dog.pawbook.logic.commands.CommandTestUtil.INVALID_TAG_DESC;
+import static dog.pawbook.logic.commands.CommandTestUtil.NAME_DESC_OBEDIENCE_TRAINING;
+import static dog.pawbook.logic.commands.CommandTestUtil.NAME_DESC_POTTY_TRAINING;
+import static dog.pawbook.logic.commands.CommandTestUtil.PREAMBLE_NON_EMPTY;
+import static dog.pawbook.logic.commands.CommandTestUtil.PREAMBLE_WHITESPACE;
+import static dog.pawbook.logic.commands.CommandTestUtil.SESSION_DESC_OBEDIENCE_TRAINING;
+import static dog.pawbook.logic.commands.CommandTestUtil.SESSION_DESC_POTTY_TRAINING;
+import static dog.pawbook.logic.commands.CommandTestUtil.TAG_DESC_PUPPIES;
+import static dog.pawbook.logic.commands.CommandTestUtil.TAG_DESC_QUIET;
+import static dog.pawbook.logic.commands.CommandTestUtil.VALID_NAME_POTTY_TRAINING;
+import static dog.pawbook.logic.commands.CommandTestUtil.VALID_SESSION_OBEDIENCE_TRAINING;
+import static dog.pawbook.logic.commands.CommandTestUtil.VALID_SESSION_POTTY_TRAINING;
+import static dog.pawbook.logic.commands.CommandTestUtil.VALID_TAG_PUPPIES;
+import static dog.pawbook.logic.commands.CommandTestUtil.VALID_TAG_QUIET;
+import static dog.pawbook.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static dog.pawbook.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static dog.pawbook.testutil.TypicalEntities.POTTY_TRAINING;
+
+import org.junit.jupiter.api.Test;
+
+import dog.pawbook.logic.commands.AddProgramCommand;
+import dog.pawbook.model.managedentity.Name;
+import dog.pawbook.model.managedentity.program.Program;
+import dog.pawbook.model.managedentity.program.Session;
+import dog.pawbook.model.managedentity.tag.Tag;
+import dog.pawbook.testutil.ProgramBuilder;
+
+public class AddProgramCommandParserTest {
+    private AddProgramCommandParser parser = new AddProgramCommandParser();
+
+    @Test
+    public void parse_allFieldsPresent_success() {
+        Program expectedProgram = new ProgramBuilder(POTTY_TRAINING).withSessions(VALID_SESSION_POTTY_TRAINING)
+            .withTags(VALID_TAG_PUPPIES).build();
```

Comment:
I think your IDE isn't configured properly, the hanging indentation should always be 8 spaces as per the java coding standard for this module.